bench/thor/routes.cc

```diff
@@ -276,7 +276,7 @@ static void BM_Sif_Allowed(benchmark::State& state) {
   // auto pred = sif::EdgeLabel(0, tgt_edge_id, edge, costs, 1.0, 1.0,
   // sif::TravelMode::kDrive,10,sif::Cost());
   auto pred = sif::EdgeLabel();
-  int restriction_idx;
+  uint8_t restriction_idx;
```

Comment:
you'll see changes similar to this throughout, basically we didnt need 2^32 values to represent the restriction index for a restriction at a particular edge. if we ever find an edge that even has 256 restrictions i'll eat my hat :smile:

Reply:
curious what the benefit is vs. the potential for overflow (though rare do you now have to do some bounds checking anywhere?)

Reply:
yeah inside the function that actually checks for restrictions (in dynamic cost) if the index is larger than 254 we cant tell what restriction was there in the route. note that this doesnt mean that we wont adhere to the restriction, it just means that the serializer doesnt know that a restriction was there.